darts/models/forecasting/prophet_model.py

```diff
@@ -318,13 +334,20 @@ def add_seasonality(
         fourier_order: int,
         prior_scale: Optional[float] = None,
         mode: Optional[str] = None,
+        condition_func: Optional[types.FunctionType] = None,
```

Comment:
I would suggest to use `condition_name` instead of a function and check at fit/predict time that the `future_covariates` series contains the boolean (binary) component `component_name`.

For this to work we also need to avoid adding the conditions as regressors in the following line: 
https://github.com/unit8co/darts/blob/b63543b8b6ec515c11fcb88cc54ad7728a6d654d/darts/models/forecasting/prophet_model.py#L200

I did a quick test and it seems to work fine.

Reply:
Moved in latest commit to use the `condition_name` and `future_covariates` instead of sending a function